client/src/canvas/components/Canvas.tsx

```diff
@@ -16,49 +16,204 @@ interface Props {
 
 class Canvas extends Component<Props> {
   canvasRef: RefObject<HTMLCanvasElement>;
+  mouseDown: Boolean;
+  widthCanvas: any;
+  heightCanvas: any;
+  xleftView: any;
+  ytopView: any;
+  widthViewOriginal: any;
+  heightViewOriginal: any;
+  widthView: any;
+  heightView: any;
+  lastX: any;
+  lastY: any;
 
   constructor(props) {
     super(props);
     this.canvasRef = createRef();
+    this.mouseDown = false;
+    this.widthCanvas = 0;
+    this.heightCanvas = 0;
+    this.xleftView = 0;
+    this.ytopView = 0;
+    this.widthViewOriginal = 1.0; //actual width and height of zoomed and panned display
+    this.heightViewOriginal = 1.0;
+    this.widthView = this.widthViewOriginal; //actual width and height of zoomed and panned display
+    this.heightView = this.heightViewOriginal;
+    this.lastX = 0;
+    this.lastY = 0;
   }
 
   componentDidMount() {
+    this.setup();
+  }
+
+  setup() {
     this.canvasRef.current!.width = 1000;
     this.canvasRef.current!.height = 1000;
+    this.widthCanvas = this.canvasRef.current!.width;
+    this.heightCanvas = this.canvasRef.current!.height;
+    const context = this.canvasRef.current!.getContext("2d");
 
-    const context = this.canvasRef.current!.getContext('2d');
-    // const image = new Image();
+    if (context) {
+      this.props.registerContext(context);
+    }
 
-    // image.onload = function() {
-    //   if (context) {
-    //     context.drawImage(image, 0, 0);
-    //   }
-    // };
-    // image.src = this.props.initialImage;
+    this.canvasRef.current!.addEventListener(
+      "dblclick",
+      this.handleDblClick,
+      false
+    ); // dblclick to zoom in at point, shift dblclick to zoom out.
+    this.canvasRef.current!.addEventListener(
+      "mousedown",
+      this.handleMouseDown,
+      false
+    ); // click and hold to pan
+    this.canvasRef.current!.addEventListener(
+      "mousemove",
+      this.handleMouseMove,
+      false
+    );
+    this.canvasRef.current!.addEventListener(
+      "mouseup",
+      this.handleMouseUp,
+      false
+    );
+    this.canvasRef.current!.addEventListener(
+      "mousewheel",
+      this.handleMouseWheel,
+      false
+    );
 
+    this.draw(context);
+  }
 
-    if (context) {
-      this.props.registerContext(context);
+  draw(ctx) {
+    ctx.setTransform(1, 0, 0, 1, 0, 0);
+    ctx.scale(
+      this.widthCanvas / this.widthView,
+      this.heightCanvas / this.heightView
+    );
+    ctx.translate(-this.xleftView, -this.ytopView);
+
+    ctx.fillStyle = "yellow";
+    ctx.fillRect(
+      this.xleftView,
+      this.ytopView,
+      this.widthView,
+      this.heightView
+    );
+    ctx.fillStyle = "blue";
+    ctx.fillRect(0.1, 0.5, 0.1, 0.1);
+    ctx.fillStyle = "red";
+    ctx.fillRect(0.3, 0.2, 0.4, 0.2);
+    ctx.fillStyle = "green";
+    ctx.beginPath();
```

Comment:
I think you could go ahead and get this to use the actual color that we choose and not these hard-coded ones as that code has been merged in.